src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1069,10 +1069,47 @@ public void GetProcessesByName_ProcessName_ReturnsExpected()
         {
             // Get the current process using its name
             Process currentProcess = Process.GetCurrentProcess();
+            Assert.NotNull(currentProcess.ProcessName);
+            Assert.NotEmpty(currentProcess.ProcessName);
 
             Process[] processes = Process.GetProcessesByName(currentProcess.ProcessName);
-            Assert.NotEmpty(processes);
+            try
+            { 
+                Assert.NotEmpty(processes);
+            }
+            catch
+            {
+                PrintProcesses();
```

Comment:
Rather than writing to the console, could we throw an exception that contains both this text and that wraps the assert exception?  Otherwise they may not be grouped together in the log.

Reply:
Good idea ok